include/seqan3/core/type_traits/iterator.hpp

```diff
@@ -30,50 +30,68 @@ namespace seqan3
 // value_type
 // ----------------------------------------------------------------------------
 
+#ifdef SEQAN3_DEPRECATED_310
+namespace detail
+{
 /*!\brief Exposes the `value_type` of another type.
  * \implements seqan3::transformation_trait
  * \tparam it_t The type you wish to query; must model std::input_iterator.
+ * \deprecate This is deprecated use std::iter_value_t.
```

Comment:
```suggestion
 * \deprecated This is deprecated use std::iter_value_t.
```

and all other occurrences.